lib/src/widgets/adw/new/window_button.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:libadwaita/src/utils/colors.dart';
+import 'package:libadwaita/src/widgets/widgets.dart';
+
+enum AdwWindowButtonType { close, maximize, minimize }
+
+class AdwWindowButton extends StatelessWidget {
+  const AdwWindowButton({
+    Key? key,
+    required this.buttonType,
+    required this.onPressed,
+    this.themeType,
+    this.windowDecor,
+  }) : super(key: key);
+
+  final dynamic themeType;
+
+  final Widget Function(
+    String name,
+    dynamic type,
+    void Function()? windowDecor,
+  )? windowDecor;
+  final VoidCallback? onPressed;
+  final AdwWindowButtonType buttonType;
+
+  @override
+  Widget build(BuildContext context) {
+    return onPressed != null
+        ? windowDecor != null
+            ? windowDecor!(buttonType.name, themeType, onPressed)
+            : AdwButton.circular(
```

Comment:
Add `margin: const EdgeInsets.all(6),` to the AdwButtons to match original behavior